### PR TITLE
wp_skill makes you avoid bad weakpoint

### PR DESCRIPTION
**Pull request by osuphobia**
#### Summary
Balance "wp_skill makes you avoid weakpoint not beneficial"
#### Purpose of change
e4e3e1c608c79b59fe05ea5f62f229fe04858568 introduced `is_good` which separates good weakpoint that is beneficial for you to hit from bad wp. I think we can make use of it.
#### Describe the solution
When calculating probability of hitting wp, use erfc if the wp does not benefit the attacker.
Now when you are good at hitting beneficial wp, you will also be tend to avoid bad wp.
#### Describe alternatives you've considered
Adjust the default difficulty of wp, but that will lead to many side effects.
#### Testing
It compiled :)
With no skill and no wp prof, the probability of hitting the body armor is 40%.
With some wp_skill, the probability decreases, and reduces to nearly zero with max gun skills and full set of wp prof.

#### Additional context
I'm working on combining critical hit and weakpoint for ranged attack, and decide to cherry-pick this feature out first.

### CODE REVIEW

**Comment by Maleclypse** [comment]
Can you share the math so with like no proficiency no gun skill =base chance to hit bad weakpoint?  Half proficiency and 5 guns what percentage of base weakpoint chance. Same for max and max.

**Comment by osuphobia** [comment]
> Can you share the math so with like no proficiency no gun skill =base chance to hit bad weakpoint? Half proficiency and 5 guns what percentage of base weakpoint chance. Same for max and max.

Yah, still take cop z as an example:

  | FALSE | TRUE
-- | -- | --
prof_gross_anatomy | 0 | 0
prof_intro_biology | 0 | 0
prof_physiology | 0 | 2
prof_wp_zombie | 0 | 2
prof_wp_syn_armored | -4 | 4

  | wp_skill | wp_skill - difficulty | difficulty_mult | raw_chance | hit_chance
-- | -- | -- | -- | -- | --
no skill no prof | -4 | -14 | 1 | 40 | 40
3 skill no prof | -1 | -11 | 0.999999981 | 39.99999924 | 40
3 skill wp_z | 1 | -9 | 0.999996602 | 39.99986409 | 49
3 skill wp_z wp_phy | 3 | -7 | 0.999767371 | 39.99069484 | 59
6 skill wp_z | 4 | -6 | 0.998650102 | 39.94600408 | 57
6 skill wp_z wp_phy | 6 | -4 | 0.977249868 | 39.08999472 | 47
8 skill wp_z wp_phy | 8 | -2 | 0.841344746 | 33.65378984 | 36
10 skill wp_z wp_phy | 10 | 0 | 0.5 | 20 | 25
3 skill all_wp | 11 | 1 | 0.308537539 | 12.34150155 | 18
6 skill all_wp | 14 | 4 | 0.022750132 | 0.910005278 | 1.8
8 skill all wp | 16 | 6 | 0.001349898 | 0.053995921 | 0.1
10 skill all wp | 18 | 8 | 3.16712E-05 | 0.00126685 | 0.003

The reweighed `hit_chance` is directly tested in game.

**Comment by Maleclypse** [comment]
I’m concerned with where it ends up at the all tens but I’m also not sure that’s a realistic concern in game balance.

**Comment by osuphobia** [comment]
Then maybe floor difficulty_mult at 10% or something for bad wp?

File: src/weakpoint.cpp [hunk 1]
```diff
@@ -535,16 +535,24 @@ float weakpoint::hit_chance( const weakpoint_attack &attack ) const
...
+        if( attack.source && attack.source->as_character() ) {
+            final_coverage = attack.source->as_character()->enchantment_cache->modify_value(
+                                 enchant_vals::mod::WEAKPOINT_ACCURACY, coverage );
+        } else {
+            final_coverage = coverage;
+        }
     } else {
+        // Use erfc if the wp does not benefit the attacker.
+        diff = attack.wp_skill - std::max( difficulty.of( attack ) + 10.0f, 10.0f );
+        difficulty_mult = 0.5f * erfc( diff / ( 2.0f * sqrt( 2.0f ) ) );
```
Comment (osuphobia) [inline src/weakpoint.cpp:555]:
```suggestion
        difficulty_mult = std::max( 0.5f * erfc( diff / ( 2.0f * sqrt( 2.0f ) ) ), 0.1f );
```

**Comment by Maleclypse** [comment]
Thank you for doing this!

**Pull request merged**